docs/TPM2_PCR_MEASUREMENTS.md

```diff
@@ -0,0 +1,137 @@
+
+If the kernel command line was specified explicitly (by the user or in a Boot
+Loader Specification Type #1 file), the kernel command line passed to the
+invoked kernel is measured before it is executed.
+
+*Description* in the event log record is the literal kernel command line in
+UTF-16.
+
+*Measured hash* covers the literal kernel command line in UTF-16 (without any
+trailing NUL byte).
+
+## PCR Measurements Made by `systemd-stub` (UEFI)
+
+### PCR 11, EV_IPL "PE Section Name"
+
+Happens once for each UKI-defined PE section of the UKI, in the canonical UKI
+PE section order, as per the UKI specification. For each record a pair of
+records are written, first one that covers the PE Section Name, and then one
+that covers the "PE Section Data", so that both types of records appear
+interleaved in the event log.
+
+*Description* in the event log record is the PE section name in UTF-16.
+
+*Measured hash* covers the PE section name in UTF-8 (including a trailing NUL byte!).
+
+### PCR 11, EV_IPL, "PE Section Data"
+
+Happens once for each UKI-defined PE section of the UKI, in the canonical UKI
+PE section order, as per the UKI specification, see above.
+
+Description in the event log record is the PE section name in UTF-16.
+
+Measured hash covers the PE section contents.
+
+### PCR 12, EV_IPL, "Kernel Command Line"
+
+Might happen up to four times, for kernel command lines from:
+
+ 1. Passed cmdline
+ 2. System cmdline add-ons (one measurement covering all add-ons combined)
+ 3. Per-UKI cmdline add-ons (one measurement covering all add-ons combined)
+ 2. SMBIOS cmdline
+
+*Description* in the event log record is the literal kernel command line in
+UTF-16.
+
+*Measured hash* covers the literal kernel command line in UTF-16 (without any
+trailing NUL byte).
+
+### PCR 12, EV_IPL, "Per-UKI Credentials initrd"
+
+*Description* in the event log record is the constant string "Credentials
+initrd" in UTF-16.
+
+*Measured hash* covers the per-UKI credentials cpio archive (which is generated
+ on-the-fly by `systemd-stub`).
+
+### PCR 12, EV_IPL, "Global Credentials initrd"
+
+*Description* in the event log record is the constant string "Global
+credentials initrd" in UTF-16.
+
+*Measured hash* covers the global credentials cpio archive (which is generated
+on-the-fly by `systemd-stub`).
+
+### PCR 13, EV_IPL, "sysext initrd"
+
+*Description* in the event log record is the constant string "System extension
+initrd" in UTF-16.
+
+*Measured hash* covers the per-UKI sysext cpio archive (which is generated
+on-the-fly by `systemd-stub`).
+
+## PCR Measurements Made by `systemd-pcrextend` (Userspace)
+
+### PCR 11, "Boot Phases"
+
+The `systemd-pcrphase.service`, `systemd-pcrphase-initrd.service`,
+`systemd-pcrphase-sysinit.service` services will measure the boot phase reached
+during various times of the boot process. Specifically, the strings
+"enter-initrd", "leave-initrd", "sysinit", "ready", "shutdown", "final" are
+measured, in this order.
```

Comment:
Maybe mention that those are normal units so the distribution or local adminstrator may modify the list the of measured phases?